v1/src/lib.rs

```diff
@@ -381,17 +380,28 @@ impl InfluxDbStorage {
         }
     }
 
-    async fn schedule_measurement_drop(&self, measurement: &str) -> TimedHandle {
-        let event = TimedEvent::once(
-            Instant::now() + Duration::from_millis(DROP_MEASUREMENT_TIMEOUT_MS),
-            TimedMeasurementDrop {
-                client: self.admin_client.clone(),
-                measurement: measurement.to_string(),
-            },
-        );
-        let handle = event.get_handle();
-        self.timer.add_async(event).await;
-        handle
+    async fn schedule_measurement_drop(&self, measurement: &str) {
+        let m_string = measurement.to_string();
+        let cloned_client = self.client.clone();
+
+        // Wait till timeout expires and execute drop,
+        // When this plugin executes as a dynamically loaded plugin,
+        // the zenohd tokio-runtime is unavailable, therefore the plugin's runtime must be invoked
+        let async_drop = async {
+            if let Err(_) = tokio::time::timeout(
```

Comment:
Couldn't this be replaced simply by a `tokio::time::sleep?`. In this particular case the timeout will always expire.

Reply:
yes thats cleaner. implemented.